src/packet_analysis/protocol/iptunnel/IPTunnel.h

```diff
@@ -78,6 +78,27 @@ class IPTunnelAnalyzer : public Analyzer
 	IPTunnelMap ip_tunnels;
 	};
 
+/**
+ * Utility function for packet analyzers for tunnel protocols. This builds a new
+ * packet and sets up the encapsulation data based on the provided information.
+ *
+ * TODO: Comments here
```

Comment:
The intention is just to add a method that creates the inner packet object and adds to the encapsulation stack in a single place so that the analyzers don't have to duplicate that code every time. I added comments to it, and made the method return the new packet instead of having the caller pass it in.

Reply:
Ideally, I'd like to deprecate/remove the other `ProcessEncapsulatedPacket` methods, or at least make them private, since the only code using them anymore is the IP-in-IP tunnel analyzer.